src/App.tsx

```diff
@@ -1,39 +1,69 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  interval: NodeJS.Timer | undefined;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+      this.rightMouseClick();
+    });
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', () => (
+      this.leftMouseClick()
+    ));
 
-        {' time is '}
+    if (this.state.hasClock) {
+      this.interval = setInterval(() => {
+        this.setState({ clockName: getRandomName() });
+      }, 3300);
+    }
+  }
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
-      </div>
-    </div>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.rightMouseClick);
```

Comment:
on line 25 you have (event) => {
      event.preventDefault();
      this.rightMouseClick();
    }
    
  this is your event handler, not this.rightMouseClick();